src/main/java/com/ericsson/eiffel/remrem/publish/helper/RMQHelper.java

```diff
@@ -218,6 +225,9 @@ public void shutdownCompleted(ShutdownSignalException cause) {
 
 
     private Channel giveMeRandomChannel() {
+        if ((rabbitChannels == null || !rabbitConnection.isOpen()) && !Boolean.getBoolean(PropertiesConfig.CLI_MODE)) {
```

Comment:
Why do you check if CLI or not? It is not necessary here but it might get you troubles in the future. I propose to remove the check.